File: src/utility/UpDownLeftRight.java
package utility;

public class UpDownLeftRight <X> {
	
	private final X up;
	private final X down;
	private final X left;
	private final X right;
	
	public UpDownLeftRight(X up, X down, X left, X right) {
		this.up = up;
		this.down = down;
		this.left = left;
		this.right = right;
	}

	public X getUp() {
		return up;
	}

	public X getDown() {
		return down;
	}

	public X getLeft() {
		return left;
	}

	public X getRight() {
		return right;
	}
	
	

}

File: Fabbri_Lucia/Csharp_LuciaFabbri/UpDownLeftRight.cs
using System;

namespace dynamicBody
{
    /// <summary>
    /// Class used to represents all the different dynamic body dimensions based on their images, in order to be as real as possible
    /// </summary>
    /// <typeparam name="X">  generic param X that will be replace by any type passed as a parameter </typeparam>

    public class UpDownLeftRight<X>
    {
        private readonly X up;
	    private readonly X down;
	    private readonly X left;
	    private readonly X right;

	    public UpDownLeftRight(X up, X down, X left, X right)
        {
            this.up = up;
            this.down = down;
            this.left = left;
            this.right = right;
        }

        public X GetUp()
        {
            return up;
        }

        public X GetDown()
        {
            return down;
        }

        public X GetLeft()
        {
            return left;
        }

        public X GetRight()
        {
            return right;
        }
    }
}
